### PR TITLE
fix: when queue cannot find task, expect queue runs next task

### PR DESCRIPTION
**Pull request by levibostian**
Through reviewing a set of SDK debug logs recently, I noticed the queue can get into an edge case that prevents it from ever finishing. Blocking the queue from running any more tasks while the app is in memory. 

I was able to reproduce the issue through a test and then implemented a bug fix. 

# Tasks 
- [ ] Release bug fix on wrapper SDKs

### CODE REVIEW

File: Sources/Common/Background Queue/QueueRunRequest.swift [hunk 1]
```diff
@@ -95,7 +95,7 @@ public class CioQueueRunRequest: QueueRunRequest {
 
                 // The task failed to execute like a HTTP failure. Update `lastFailedTask`.
                 updateWhileLoopLogicVariables(didTaskFail: true, taskJustExecuted: nextTaskToRunInventoryItem)
-                break
+                continue // quit loop early and move to next task.
```
Comment (ami-aman) [inline Sources/Common/Background Queue/QueueRunRequest.swift:98]: Does this possibly fix the [bug that the customer reported](https://github.com/customerio/issues/issues/11546) ?

**Review by ami-aman — APPROVED** [review]
